doc/system_requirements.md

```diff
@@ -0,0 +1,127 @@
+# System Requirement Specification Parquet IO Java
+
+## Introduction
+
+Parquet IO Java (PIOJ) is a Java library to access Apache Parquet files. Users
+can use this library to read Parquet rows into Java objects.
+
+## About This Document
+
+### Target Audience
+
+The target audience are software developers that use Java Virtual Machine (JVM)
+based programming languages. See section ["Stakeholders"](#stakeholders) for
+more details.
+
+### Goal
+
+The goal of Parquet IO Java is to provide reliable and efficient programmatic
+access to Parquet files without using distributed compute engines.
+
+## Stakeholders
+
+Software Developers use PIOJ to programmatically access Parquet files.
+
+## Terms and Abbreviations
+
+The following list gives you an overview of terms and abbreviations commonly
+used in PIOJ documents.
+
+<dl>
+<dt>Parquet</dt><dd>A columnar storage format.</dd>
+<dt>Parquet File</dt><dd>A file that contains Parquet formatted data.</dd>
+<dt>Row</dt><dd>A abstraction over list of columns in Parquet formatted data.</dd>
+</dl>
+
+## Features
+
+Features are the highest level requirements in this document that describe the
+main functionality of Parquet IO Java.
+
+### Parquet Rows 
+`feat~parquet-rows~1`
+
+PIOJ allows user to access Parquet rows as Java objects.
+
+Needs: req
+
+### Column Data Reading
```

Comment:
I think it is better as two separate requirement.
The first one is more about providing easy access to the decoded data, e.g, positional or using column names.

Second is a feature that it can read all parquet types of a column.